BUG: Unable to use scipy development environment from IDE

### Describe your issue.

After following the steps to build `scipy` from source (https://scipy.github.io/devdocs/building/index.html#building-from-source):
```
mamba env create -f environment.yml
mamba activate scipy-dev
python dev.py build
```
I was unable to use the environment in the IDE (e.g. in the Jupyter Interactive Window in VSCode), as the `scipy` package cannot be imported:
```ModuleNotFoundError: No module named 'scipy'```

`scipy` is not present in the list of packages in the environment:
```conda list | grep scipy```

Presumably, this might be fixed once support for editable installs is added (https://github.com/scipy/scipy/issues/16293).

### Reproducing Code Example

```python
import scipy
```


### Error message

```shell
ModuleNotFoundError: No module named 'scipy'
```


### SciPy/NumPy/Python version and system information

```shell
1.12.0.dev0+1341.4172aee 1.25.1 sys.version_info(major=3, minor=10, micro=12, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /Users/artem/micromamba/envs/scipy-dev/include
    lib directory: /Users/artem/micromamba/envs/scipy-dev/lib
    name: openblas
    openblas configuration: USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK=0 NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=1 CORE2 MAX_THREADS=128
    pc file directory: /Users/artem/micromamba/envs/scipy-dev/lib/pkgconfig
    version: 0.3.23
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /Users/artem/micromamba/envs/scipy-dev/include
    lib directory: /Users/artem/micromamba/envs/scipy-dev/lib
    name: openblas
    openblas configuration: USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK=0 NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=1 CORE2 MAX_THREADS=128
    pc file directory: /Users/artem/micromamba/envs/scipy-dev/lib/pkgconfig
    version: 0.3.23
  pybind11:
    detection method: pkgconfig
    include directory: /Users/artem/micromamba/envs/scipy-dev/include
    name: pybind11
    version: 2.11.0
Compilers:
  c:
    args: -march=core2, -mtune=haswell, -mssse3, -ftree-vectorize, -fPIC, -fPIE, -fstack-protector-strong,
      -O2, -pipe, -isystem, /Users/artem/micromamba/envs/scipy-dev/include, -D_FORTIFY_SOURCE=2,
      -isystem, /Users/artem/micromamba/envs/scipy-dev/include
    commands: x86_64-apple-darwin13.4.0-clang
    linker: ld64
    linker args: -Wl,-pie, -Wl,-headerpad_max_install_names, -Wl,-dead_strip_dylibs,
      -Wl,-rpath,/Users/artem/micromamba/envs/scipy-dev/lib, -L/Users/artem/micromamba/envs/scipy-dev/lib,
      -march=core2, -mtune=haswell, -mssse3, -ftree-vectorize, -fPIC, -fPIE, -fstack-protector-strong,
      -O2, -pipe, -isystem, /Users/artem/micromamba/envs/scipy-dev/include, -D_FORTIFY_SOURCE=2,
      -isystem, /Users/artem/micromamba/envs/scipy-dev/include
    name: clang
    version: 15.0.7
  c++:
    args: -march=core2, -mtune=haswell, -mssse3, -ftree-vectorize, -fPIC, -fPIE, -fstack-protector-strong,
      -O2, -pipe, -stdlib=libc++, -fvisibility-inlines-hidden, -fmessage-length=0,
      -isystem, /Users/artem/micromamba/envs/scipy-dev/include, -D_FORTIFY_SOURCE=2,
      -isystem, /Users/artem/micromamba/envs/scipy-dev/include
    commands: x86_64-apple-darwin13.4.0-clang++
    linker: ld64
    linker args: -Wl,-pie, -Wl,-headerpad_max_install_names, -Wl,-dead_strip_dylibs,
      -Wl,-rpath,/Users/artem/micromamba/envs/scipy-dev/lib, -L/Users/artem/micromamba/envs/scipy-dev/lib,
      -march=core2, -mtune=haswell, -mssse3, -ftree-vectorize, -fPIC, -fPIE, -fstack-protector-strong,
      -O2, -pipe, -stdlib=libc++, -fvisibility-inlines-hidden, -fmessage-length=0,
      -isystem, /Users/artem/micromamba/envs/scipy-dev/include, -D_FORTIFY_SOURCE=2,
      -isystem, /Users/artem/micromamba/envs/scipy-dev/include
    name: clang
    version: 15.0.7
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 0.29.36
  fortran:
    args: -march=core2, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector,
      -O2, -pipe, -isystem, /Users/artem/micromamba/envs/scipy-dev/include
    commands: /Users/artem/micromamba/envs/scipy-dev/bin/x86_64-apple-darwin13.4.0-gfortran
    linker: ld64
    linker args: -Wl,-pie, -Wl,-headerpad_max_install_names, -Wl,-dead_strip_dylibs,
      -Wl,-rpath,/Users/artem/micromamba/envs/scipy-dev/lib, -L/Users/artem/micromamba/envs/scipy-dev/lib,
      -march=core2, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector, -O2,
      -pipe, -isystem, /Users/artem/micromamba/envs/scipy-dev/include
    name: gcc
    version: 12.2.0
  pythran:
    include directory: ../../../../micromamba/envs/scipy-dev/lib/python3.10/site-packages/pythran
    version: 0.13.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: darwin
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: darwin
Python Information:
  path: /Users/artem/micromamba/envs/scipy-dev/bin/python3.10
  version: '3.10'
```

> `cd doc && make html`

That is giving me:
```
scipy not found, cannot build documentation without successful "import scipy"
make: *** [Makefile:60: version-check] Error 1
```

> Then your editable install did not work, or you are in the wrong environment. Or it's picking up the wrong Sphinx. If the former, then python -c "import scipy" will fail with the same message. If the latter, please install Sphinx and the other doc dependencies in your active env.

That was my bad - I was trying to run it from the wrong environment.

Once in the editable environment, I got this error:

```
# for testing
# @echo installed scipy aad99b4 matches git version aad99b4; exit 1
mkdir -p build/html build/doctrees
LANG=C python3 -msphinx -WT --keep-going  -b html -d build/doctrees  source build/html 
Running Sphinx v5.3.0
SciPy (VERSION 1.12.0.dev)
/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/util/images.py:4: DeprecationWarning: 'imghdr' is deprecated and slated for removal in Python 3.13
  import imghdr

Traceback (most recent call last):
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/application.py", line 223, in __init__
    self.setup_extension(extension)
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/application.py", line 398, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/registry.py", line 472, in load_extension
    metadata = setup(app)
               ^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/__init__.py", line 8, in setup
    from .sphinx_ext import sphinx_setup
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/sphinx_ext.py", line 20, in <module>
    from myst_nb.ext.eval import load_eval_sphinx
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/ext/eval/__init__.py", line 10, in <module>
    from myst_nb.core.execute.base import EvalNameError
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/core/execute/__init__.py", line 7, in <module>
    from .cache import NotebookClientCache
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/core/execute/cache.py", line 13, in <module>
    from jupyter_cache.executors.utils import single_nb_execution
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 7, in <module>
    from nbclient import execute as executenb
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/nbclient/__init__.py", line 5, in <module>
    from .client import NotebookClient, execute  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/nbclient/client.py", line 15, in <module>
    from jupyter_client import KernelManager
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_client/__init__.py", line 3, in <module>
    from .asynchronous import AsyncKernelClient
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_client/asynchronous/__init__.py", line 1, in <module>
    from .client import AsyncKernelClient  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_client/asynchronous/client.py", line 9, in <module>
    from ..client import KernelClient, reqrep
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_client/client.py", line 20, in <module>
    from .connect import ConnectionFileMixin
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_client/connect.py", line 20, in <module>
    from jupyter_core.paths import jupyter_data_dir, jupyter_runtime_dir, secure_write
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_core/paths.py", line 208, in <module>
    deprecation(
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 89, in deprecation
    warnings.warn(message, DeprecationWarning, stacklevel=stacklevel + 1)
DeprecationWarning: Jupyter is migrating its paths to use standard platformdirs
given by the platformdirs library.  To remove this warning and
see the appropriate new directories, set the environment variable
`JUPYTER_PLATFORM_DIRS=1` and then run `jupyter --paths`.
The use of platformdirs will be the default in `jupyter_core` v6

Exception occurred:
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 89, in deprecation
    warnings.warn(message, DeprecationWarning, stacklevel=stacklevel + 1)
DeprecationWarning: Jupyter is migrating its paths to use standard platformdirs
given by the platformdirs library.  To remove this warning and
see the appropriate new directories, set the environment variable
`JUPYTER_PLATFORM_DIRS=1` and then run `jupyter --paths`.
The use of platformdirs will be the default in `jupyter_core` v6
The full traceback has been saved in /tmp/sphinx-err-ba_2e_7a.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:110: html-build] Error 2
```

Setting `JUPYTER_PLATFORM_DIRS=1` seems to have solved it, but then it hit the next issue:

```
# for testing
# @echo installed scipy aad99b4 matches git version aad99b4; exit 1
mkdir -p build/html build/doctrees
LANG=C python3 -msphinx -WT --keep-going  -b html -d build/doctrees  source build/html 
Running Sphinx v5.3.0
SciPy (VERSION 1.12.0.dev)
/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/util/images.py:4: DeprecationWarning: 'imghdr' is deprecated and slated for removal in Python 3.13
  import imghdr

Traceback (most recent call last):
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/application.py", line 223, in __init__
    self.setup_extension(extension)
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/application.py", line 398, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/sphinx/registry.py", line 472, in load_extension
    metadata = setup(app)
               ^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/__init__.py", line 10, in setup
    return sphinx_setup(app)
           ^^^^^^^^^^^^^^^^^
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/sphinx_ext.py", line 98, in sphinx_setup
    add_css(app)
  File "/home/artem/git/scipy/venv/lib/python3.11/site-packages/myst_nb/sphinx_ext.py", line 189, in add_css
    with import_resources.path(static, "mystnb.css") as source_path:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/resources/_legacy.py", line 18, in wrapper
    warnings.warn(
DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.

Exception occurred:
  File "/usr/local/lib/python3.11/importlib/resources/_legacy.py", line 18, in wrapper
    warnings.warn(
DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
The full traceback has been saved in /tmp/sphinx-err-6i_y6hfv.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:110: html-build] Error 2
```

> Standard pytest usage. E.g. to run all tests, `pytest scipy`, and to run only cluster tests, `pytest scipy/cluster`.

I've tried running all tests from source root using `pytest scipy`, but got these three errors:

```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.11.3, pytest-7.4.0, pluggy-1.2.0
rootdir: /home/artem/git/scipy
configfile: pytest.ini
plugins: xdist-3.3.1, timeout-2.1.0
collected 58027 items / 3 errors / 1 skipped                                                                                                                                                                      

===================================================================================================== ERRORS ======================================================================================================
_________________________________________________________________________ ERROR collecting scipy/_build_utils/tests/test_scipy_version.py _________________________________________________________________________
ImportError while importing test module '/home/artem/git/scipy/scipy/_build_utils/tests/test_scipy_version.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'scipy._build_utils'
_____________________________________________________________ ERROR collecting scipy/_lib/highs/src/interfaces/highspy/highspy/tests/test_highspy.py ______________________________________________________________
ImportError while importing test module '/home/artem/git/scipy/scipy/_lib/highs/src/interfaces/highspy/highspy/tests/test_highspy.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scipy/_lib/highs/src/interfaces/highspy/highspy/__init__.py:1: in <module>
    from .highs import (
scipy/_lib/highs/src/interfaces/highspy/highspy/highs.py:1: in <module>
    from .highs_bindings import (
E   ModuleNotFoundError: No module named 'highspy.highs_bindings'
________________________________________________________________________________ ERROR collecting scipy/special/tests/test_data.py ________________________________________________________________________________
.venv/lib/python3.11/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
        cls        = <class '_pytest.runner.CallInfo'>
        duration   = 0.007315864000702277
        excinfo    = <ExceptionInfo FileNotFoundError(2, 'No such file or directory') tblen=17>
        func       = <function pytest_make_collect_report.<locals>.<lambda> at 0x7f5300662200>
        precise_start = 28024.274867782
        precise_stop = 28024.282183646
        reraise    = None
        result     = None
        start      = 1690886191.0433958
        stop       = 1690886191.050713
        when       = 'collect'
.venv/lib/python3.11/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
        collector  = <Module test_data.py>
.venv/lib/python3.11/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
        __class__  = <class '_pytest.python.Module'>
        self       = <Module test_data.py>
.venv/lib/python3.11/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
        has_nose   = True
        self       = <Module test_data.py>
.venv/lib/python3.11/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
        obj        = None
        self       = <Module test_data.py>
.venv/lib/python3.11/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
        self       = <Module test_data.py>
.venv/lib/python3.11/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
        importmode = 'prepend'
        self       = <Module test_data.py>
.venv/lib/python3.11/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
        mode       = <ImportMode.prepend: 'prepend'>
        module_name = 'scipy.special.tests.test_data'
        names      = ['scipy', 'special', 'tests', 'test_data']
        p          = PosixPath('/home/artem/git/scipy/scipy/special/tests/test_data.py')
        path       = PosixPath('/home/artem/git/scipy/scipy/special/tests/test_data.py')
        pkg_path   = PosixPath('/home/artem/git/scipy/scipy')
        pkg_root   = PosixPath('/home/artem/git/scipy')
        root       = PosixPath('/home/artem/git/scipy')
/usr/local/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
        level      = 0
        name       = 'scipy.special.tests.test_data'
        package    = None
<frozen importlib._bootstrap>:1206: in _gcd_import
    ???
        level      = 0
        name       = 'scipy.special.tests.test_data'
        package    = None
<frozen importlib._bootstrap>:1178: in _find_and_load
    ???
        import_    = <function _gcd_import at 0x7f530ca87d80>
        module     = <object object at 0x7f530cabc050>
        name       = 'scipy.special.tests.test_data'
<frozen importlib._bootstrap>:1149: in _find_and_load_unlocked
    ???
        child      = 'test_data'
        import_    = <function _gcd_import at 0x7f530ca87d80>
        name       = 'scipy.special.tests.test_data'
        parent     = 'scipy.special.tests'
        parent_module = <module 'scipy.special.tests' from '/home/artem/git/scipy/scipy/special/tests/__init__.py'>
        parent_spec = ModuleSpec(name='scipy.special.tests', loader=<_scipy_editable_loader.SourceFileLoader object at 0x7f52d3c533d0>, origin='/home/artem/git/scipy/scipy/special/tests/__init__.py', submodule_search_locations=[])
        path       = []
        spec       = ModuleSpec(name='scipy.special.tests.test_data', loader=<_scipy_editable_loader.SourceFileLoader object at 0x7f52d3603810>, origin='/home/artem/git/scipy/scipy/special/tests/test_data.py')
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
        module     = <module 'scipy.special.tests.test_data' from '/home/artem/git/scipy/scipy/special/tests/test_data.py'>
        spec       = ModuleSpec(name='scipy.special.tests.test_data', loader=<_scipy_editable_loader.SourceFileLoader object at 0x7f52d3603810>, origin='/home/artem/git/scipy/scipy/special/tests/test_data.py')
<frozen importlib._bootstrap_external>:940: in exec_module
    ???
        code       = <code object <module> at 0x55bd2633f740, file "/home/artem/git/scipy/scipy/special/tests/test_data.py", line 1>
        module     = <module 'scipy.special.tests.test_data' from '/home/artem/git/scipy/scipy/special/tests/test_data.py'>
        self       = <_scipy_editable_loader.SourceFileLoader object at 0x7f52d3603810>
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
        args       = (<code object <module> at 0x55bd2633f740, file "/home/artem/git/scipy/scipy/special/tests/test_data.py", line 1>, {'Fu...ception'>, ...}, '__cached__': '/home/artem/git/scipy/scipy/special/tests/__pycache__/test_data.cpython-311.pyc', ...})
        f          = <built-in function exec>
        kwds       = {}
scipy/special/tests/test_data.py:26: in <module>
    DATASETS_BOOST = np.load(os.path.join(os.path.dirname(__file__),
        FuncData   = <class 'scipy.special._testutils.FuncData'>
        IntegrationWarning = <class 'scipy.integrate._quadpack_py.IntegrationWarning'>
        __builtins__ = <builtins>
        __cached__ = '/home/artem/git/scipy/scipy/special/tests/__pycache__/test_data.cpython-311.pyc'
        __doc__    = None
        __file__   = '/home/artem/git/scipy/scipy/special/tests/test_data.py'
        __loader__ = <_scipy_editable_loader.SourceFileLoader object at 0x7f52d3603810>
        __name__   = 'scipy.special.tests.test_data'
        __package__ = 'scipy.special.tests'
        __spec__   = ModuleSpec(name='scipy.special.tests.test_data', loader=<_scipy_editable_loader.SourceFileLoader object at 0x7f52d3603810>, origin='/home/artem/git/scipy/scipy/special/tests/test_data.py')
        bdtrik     = <ufunc 'bdtrik'>
        beta       = <ufunc 'beta'>
        betainc    = <ufunc 'betainc'>
        betaincinv = <ufunc 'betaincinv'>
        binom      = <ufunc 'binom'>
        btdtr      = <ufunc 'btdtr'>
        btdtri     = <ufunc 'btdtri'>
        btdtria    = <ufunc 'btdtria'>
        btdtrib    = <ufunc 'btdtrib'>
        cbrt       = <ufunc 'cbrt'>
        chndtr     = <ufunc 'chndtr'>
        digamma    = <ufunc 'psi'>
        ellip_harm = <function ellip_harm at 0x7f52e37367a0>
        ellip_harm_2 = <function ellip_harm_2 at 0x7f52e3736700>
        ellipe     = <ufunc 'ellipe'>
        ellipeinc  = <ufunc 'ellipeinc'>
        ellipk     = <ufunc 'ellipk'>
        ellipkinc  = <ufunc 'ellipkinc'>
        ellipkm1   = <ufunc 'ellipkm1'>
        elliprc    = <ufunc 'elliprc'>
        elliprd    = <ufunc 'elliprd'>
        elliprf    = <ufunc 'elliprf'>
        elliprg    = <ufunc 'elliprg'>
        elliprj    = <ufunc 'elliprj'>
        erf        = <ufunc 'erf'>
        erfc       = <ufunc 'erfc'>
        erfcinv    = <ufunc 'erfcinv'>
        erfinv     = <ufunc 'erfinv'>
        eval_genlaguerre = <ufunc 'eval_genlaguerre'>
        eval_hermite = <ufunc 'eval_hermite'>
        eval_laguerre = <ufunc 'eval_laguerre'>
        eval_legendre = <ufunc 'eval_legendre'>
        exp1       = <ufunc 'exp1'>
        expi       = <ufunc 'expi'>
        expm1      = <ufunc 'expm1'>
        expn       = <ufunc 'expn'>
        gamma      = <ufunc 'gamma'>
        gammainc   = <ufunc 'gammainc'>
        gammaincc  = <ufunc 'gammaincc'>
        gammainccinv = <ufunc 'gammainccinv'>
        gammaincinv = <ufunc 'gammaincinv'>
        gammaln    = <ufunc 'gammaln'>
        gdtr       = <ufunc 'gdtr'>
        gdtrc      = <ufunc 'gdtrc'>
        gdtrib     = <ufunc 'gdtrib'>
        gdtrix     = <ufunc 'gdtrix'>
        iv         = <ufunc 'iv'>
        ivp        = <function ivp at 0x7f52e36f2700>
        jn         = <ufunc 'jv'>
        jv         = <ufunc 'jv'>
        jvp        = <function jvp at 0x7f52e36f2520>
        kn         = <ufunc 'kn'>
        kv         = <ufunc 'kv'>
        kvp        = <function kvp at 0x7f52e36f2660>
        log1p      = <ufunc 'log1p'>
        lpmn       = <function lpmn at 0x7f52e36f2f20>
        lpmv       = <ufunc 'lpmv'>
        lpn        = <function lpn at 0x7f52e36f3240>
        lqmn       = <function lqmn at 0x7f52e36f3060>
        lqn        = <function lqn at 0x7f52e36f32e0>
        mathieu_a  = <ufunc 'mathieu_a'>
        mathieu_b  = <ufunc 'mathieu_b'>
        mathieu_cem = <ufunc 'mathieu_cem'>
        mathieu_modcem1 = <ufunc 'mathieu_modcem1'>
        mathieu_modcem2 = <ufunc 'mathieu_modcem2'>
        mathieu_modsem1 = <ufunc 'mathieu_modsem1'>
        mathieu_modsem2 = <ufunc 'mathieu_modsem2'>
        mathieu_sem = <ufunc 'mathieu_sem'>
        nbdtrik    = <ufunc 'nbdtrik'>
        np         = <module 'numpy' from '/home/artem/git/scipy/.venv/lib/python3.11/site-packages/numpy/__init__.py'>
        os         = <module 'os' (frozen)>
        owens_t    = <ufunc 'owens_t'>
        pdtrik     = <ufunc 'pdtrik'>
        poch       = <ufunc 'poch'>
        pytest     = <module 'pytest' from '/home/artem/git/scipy/.venv/lib/python3.11/site-packages/pytest/__init__.py'>
        sph_harm   = <ufunc 'sph_harm'>
        spherical_jn = <function spherical_jn at 0x7f52e3736ac0>
        spherical_yn = <function spherical_yn at 0x7f52e3736980>
        suppress_warnings = <class 'numpy.testing._private.utils.suppress_warnings'>
        wright_bessel = <ufunc 'wright_bessel'>
        yn         = <ufunc 'yn'>
        yv         = <ufunc 'yv'>
        yvp        = <function yvp at 0x7f52e36f25c0>
        zeta       = <function zeta at 0x7f52e3734400>
.venv/lib/python3.11/site-packages/numpy/lib/npyio.py:427: in load
    fid = stack.enter_context(open(os_fspath(file), "rb"))
E   FileNotFoundError: [Errno 2] No such file or directory: '/home/artem/git/scipy/scipy/special/tests/data/boost.npz'
        allow_pickle = False
        encoding   = 'ASCII'
        file       = '/home/artem/git/scipy/scipy/special/tests/data/boost.npz'
        fix_imports = True
        max_header_size = 10000
        mmap_mode  = None
        pickle_kwargs = {'encoding': 'ASCII', 'fix_imports': True}
        stack      = <contextlib.ExitStack object at 0x7f52d3603850>
============================================================================================= short test summary info =============================================================================================
ERROR scipy/_build_utils/tests/test_scipy_version.py
ERROR scipy/_lib/highs/src/interfaces/highspy/highspy/tests/test_highspy.py
ERROR scipy/special/tests/test_data.py - FileNotFoundError: [Errno 2] No such file or directory: '/home/artem/git/scipy/scipy/special/tests/data/boost.npz'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================================================================== 1 skipped, 3 errors in 15.29s ==========================================================================================
```

In my case, the last error is
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/artem/git/scipy/scipy/special/tests/data/boost.npz'
```
so it seems to be looking for the file in the source directory rather than the build directory.

There are also there two errors:
```
E   ModuleNotFoundError: No module named 'scipy._build_utils'
E   ModuleNotFoundError: No module named 'highspy.highs_bindings'
```

Let me try running just the tests in `qmc` though.

--- a/doc/source/building/index.rst
+++ b/doc/source/building/index.rst
@@ -379,29 +379,33 @@ interface is self-documenting, so please see ``python dev.py --help`` and
 .. admonition:: IDE support & editable installs
 
     While the ``dev.py`` interface is our recommended way of working on SciPy,
-    it has one limitation: because of the custom install location, setting
-    breakpoints inside an IDE or debugger does not work as expected. This will
-    only work with an *in-place build* (or "editable install").
+    it has one limitation: because of the custom install location, SciPy
+    installed using ``dev.py`` will not be recognized automatically within an
+    IDE (e.g., for running a script via a "run" button, or setting breakpoints
+    visually). This will work better with an *in-place build* (or "editable
+    install").
 
     Editable installs are supported. It is important to understand that **you
-    must use either an editable install or dev.py, but not both**. If you use
-    editable installs, you can use ``pytest`` and other development tools
-    directly instead of using ``dev.py``.
+    may use either an editable install or dev.py in a given repository clone,
+    but not both**. If you use editable installs, you have to use ``pytest``
+    and other development tools directly instead of using ``dev.py``.
 
     To use an editable install, ensure you start from a clean repository (run
-    ``git clean -xdf`` if you've built with ``dev.py`` before) and have all dependencies set up correctly as described higher up on this page. Then do::
+    ``git clean -xdf`` if you've built with ``dev.py`` before) and have all
+    dependencies set up correctly as described higher up on this page. Then
+    do::
 
         # Note: the --no-build-isolation is important! meson-python will
-        # auto-rebuild each time Scipy is imported by the Python interpreter.
+        # auto-rebuild each time SciPy is imported by the Python interpreter.
         pip install -e . --no-build-isolation
 
         # To run the tests for, e.g., the `scipy.linalg` module:
         pytest scipy/linalg
 
     When making changes to SciPy code, including to compiled code, there is no
-    need to manually rebuild. When you run ``git clean -xdf``, which removes
-    the built extension modules, remember to also uninstall SciPy with ``pip
-    uninstall scipy``.
+    need to manually rebuild or reinstall. When you run ``git clean -xdf``,
+    which removes the built extension modules, remember to also uninstall SciPy
+    with ``pip uninstall scipy``.
 
     See the meson-python_ documentation on editable installs for more details
     on how things work under the hood.

--- a/doc/source/building/index.rst
+++ b/doc/source/building/index.rst
@@ -246,7 +246,7 @@ Building from source to use SciPy
       mamba env create -f environment.yml
 
       # Or, install only the required build dependencies
-      mamba install python numpy cython pythran pybind11 compilers openblas pkg-config
+      mamba install python numpy cython pythran pybind11 compilers openblas meson-python pkg-config
 
       # To build the latest stable release:
       pip install scipy --no-build-isolation --no-binary scipy
@@ -353,7 +353,7 @@ virtual environments:
     PyPI with::
 
        # Build dependencies
-       python -m pip install numpy cython pythran pybind11 meson ninja pydevtool rich-click
+       python -m pip install numpy cython pythran pybind11 meson-python ninja pydevtool rich-click
 
        # Test and optional runtime dependencies
        python -m pip install pytest pytest-xdist pytest-timeout pooch threadpoolctl asv gmpy2 mpmath hypothesis

--- a/doc/source/building/index.rst
+++ b/doc/source/building/index.rst
@@ -376,6 +376,37 @@ interface is self-documenting, so please see ``python dev.py --help`` and
 ``python dev.py <subcommand> --help`` for detailed guidance.
 
 
+.. admonition:: IDE support & editable installs
+
+    While the ``dev.py`` interface is our recommended way of working on SciPy,
+    it has one limitation: because of the custom install location, setting
+    breakpoints inside an IDE or debugger does not work as expected. This will
+    only work with an *in-place build* (or "editable install").
+
+    Editable installs are supported. It is important to understand that **you
+    must use either an editable install or dev.py, but not both**. If you use
+    editable installs, you can use ``pytest`` and other development tools
+    directly instead of using ``dev.py``.
+
+    To use an editable install, ensure you start from a clean repository (run
+    ``git clean -xdf`` if you've built with ``dev.py`` before) and have all dependencies set up correctly as described higher up on this page. Then do::
+
+        # Note: the --no-build-isolation is important! meson-python will
+        # auto-rebuild each time Scipy is imported by the Python interpreter.
+        pip install -e . --no-build-isolation
+
+        # To run the tests for, e.g., the `scipy.linalg` module:
+        pytest scipy/linalg
+
+    When making changes to SciPy code, including to compiled code, there is no
+    need to manually rebuild. When you run ``git clean -xdf``, which removes
+    the built extension modules, remember to also uninstall SciPy with ``pip
+    uninstall scipy``.
+
+    See the meson-python_ documentation on editable installs for more details
+    on how things work under the hood.
+
+
 Customizing builds
 ------------------
 
@@ -400,3 +431,4 @@ Background information
 
 
 .. _Mambaforge: https://github.com/conda-forge/miniforge#mambaforge
+.. _meson-python: https://mesonbuild.com/meson-python/

